src/random-seed/random-seed.c

```diff
@@ -242,6 +245,16 @@ static int run(int argc, char *argv[]) {
                         if (r < 0)
                                 log_error_errno(r, "Failed to write seed to /dev/urandom: %m");
                 }
+                /* If we're going to later write out a seed file, initialize a hash state with
+                 * the contents of the seed file we just read, so that the new one can't regress
+                 * in entropy. */
+                if (write_seed_file) {
+                        sha256_init_ctx(&hash_state);
+                        if (k < 0)
+                                k = 0;
+                        sha256_process_bytes(&k, sizeof(k), &hash_state);
+                        sha256_process_bytes(buf, k, &hash_state);
+                }
```

Comment:
this if block is unnecessary unless we actually managed to read a seed off disk, no? hence, move this into the if block immediately preceeding this, i.e. after the successful random_write_entropy()? if so, you can drop the `if (k < 0) k = 0)` thing, because that's guaranteed then.

Reply:
I figured it was cleanest to unconditionally initialize the hash state here, so that regardless of the outcome it'd be in a good state later on down when it's used again. Instead, I guess more code (really, another boolean) could be added to determine whether there is an old seed hashed or not, if you really wanted. Doesn't seem to get us anything though?

Reply:
i must say I am typically not a fan of such "implicit dependencies". people move code around, and it is not obvious under which circumstances `hash_state` is initialized or not, hence this might get out of sync by accident, in some error case or so. The whole function is unusually large already after all, hence such implicit deps are not easy to see. Hence usually i'd add an explicit bool here, `hash_state_initialized` or so, and then always set/check that. the compiler is then free to optimize it away if it wants to, but given this is not a hot path, it doesn't really matter... 

i mean, this doesn't matter much i guess, it's just something i noticed. at the very least could use a comment, so that this isn't changed by accident so easily.

Reply:
Will do.